### PR TITLE
Add module for `pbkdf2`

### PR DESCRIPTION
**Pull request by chm-diederichs**

### CODE REVIEW

**Comment by chm-diederichs** [comment]
depends on #193

File: macros.h [hunk 1]
```diff
@@ -88,8 +88,13 @@
 
 #define SN_EXPORT_UINT64(name, num) \
   { \
+    if (num < 0) { \
```
Comment (chm-diederichs) [inline macros.h]: this check triggers a `-Wtype-limits` warning when building, but shall we keep it as a sanity check?

Comment (mafintosh) [reply macros.h]: let me check

Comment (mafintosh) [reply macros.h]: would remove, doesn't make sense since we are the only users of this api

**Pull request merged**